README.md
McCall Lab Theme
================
Zachary Brehm

## Installation

This package can be installed using devtools.

``` r
devtools::install_github("zachbrehm/ggmnmlab")
```

## Introduction

This package is currently in development. This package is intended to
provide an easily deployable color palette and theme for ggplot2 for use
by students in the McCall Lab in the Department of Biostatistics at
University of Rochester. The palettes in this package are all intended
to be color blind friendly.

Supports color palettes with 8, 12, 15, or 48 color options. These
palettes were adapted by [Martin
Krzywinski](http://mkweb.bcgsc.ca/colorblind/palettes.mhtml#page-container),
of which the 8 color palette was adapted from [Points of view: Color
Blindness](https://www.nature.com/articles/nmeth.1618) by Bang Wong.
Further updates to this package will include the remaining palettes
displayed on.

``` r
library(ggmnmlab)
#> Loading required package: ggplot2
#> Loading required package: scales
library(palmerpenguins)
```

## Example plots

An example of the 8 color palette and theme is shown here, applied to
the cell-types data from <https://github.com/zachbrehm/celltypes>.

``` r
ggplot(tsneDf, aes(x = X, y = Y, color = Z, text = Project, alpha = 0.8)) + 
  geom_point() + 
  labs(title = "Cell-type data with dimension reduction by TSNE", 
       x = "First Dimension", 
       y = "Second Dimension", 
       color = "Cell-Type",
       text = rse$study) + 
  guides(alpha = FALSE) + 
  scale_color_mnmlab(palette = "martink8",
                         labels = c("Cardiac Muscle", 
                                   "Endothelial", 
                                   "Erythrocyte", 
                                   "Adipocyte", 
                                   "Lymphocyte", 
                                   "Macrophage", 
                                   "Smooth Muscle"),
                        ) +
  theme_mnmlab()
```

![](man/figures/gg_tsne.png)<!-- -->

As another example, we recreate the **Bill length vs. depth** example
from the `palmerpenguins` article, this time using the high contrast
palette from Paul Tol:
<https://allisonhorst.github.io/palmerpenguins/articles/examples.html>

``` r
ggplot(
  penguins,
  aes(x = bill_length_mm, y = bill_depth_mm, color = species, group = species)
) +
  geom_point(
    shape = 19, size = 2
  ) +
  geom_smooth(method = "lm", se = FALSE) +
  scale_color_mnmlab(palette = "highContrast") +
  labs(
    title = "Penguin bill dimensions at Palmer Station",
    x = "Bill length (mm)", y = "Bill depth (mm)", color = "Species"
  ) +
  theme_mnmlab()
```

![](man/figures/penguins.png)<!-- -->


NEWS.md
#v0.2.1
* Add palettes from Paul Tol.

# v0.2.0
* Changed the name of the package from ggmccalllab to ggmnmlab.

# v0.1.2
* Added 12, 15, and 48 color palettes.
* Updated scale_color_mccall_lab to take a palette argument to select which 
  palette to use.

# v0.1.1
* 8 color palette available


R/scale_mnmlab.R
#' Color palette for McCall Lab plots

#' 8 color scale created by [Martin Krzywinski's color blind palettes](http://mkweb.bcgsc.ca/colorblind/palettes.mhtml#page-container), adapted from [Wong, B. (2011) Nature Methods 8:441](https://www.nature.com/articles/nmeth.1618).
#' @references
#' Krzywinski, M. Designing for Color Blindess - Martin Krzywinski / Genome Sciences Center. http://mkweb.bcgsc.ca/colorblind/index.mhtml#page-container
#' Wong, B. (2011). Color blindness. In Nature Methods (Vol. 8, Issue 6, p. 441). https://doi.org/10.1038/nmeth.1618

#' @seealso [ggplot2::scale_color_discrete][ggplot2::scale_fill_discrete]
#' @inheritParams ggplot2::discrete_scale
#' @inheritParams scales::manual_pal
#' @param palette= Palette for color scale. Takes on the values martink8,
#'                 martink12, martink15, or martink48 for palettes by Martin K.
#'                 Alternatively, highContrast, bright, vibrant, muted, pale,
#'                 dark, or light can be supplied for palettes by Paul Tol.
#' @param ... Further arguments passed to other methods
#' @name scale_mnmlab
#'

mnmlab_pal <- function(palette) {
  if(startsWith(palette, "martink")){
    pal <- martink_pals[[palette]]
  }
  else{
    pal <- paulTol_pals[[palette]]
  }
  scales::manual_pal(pal)
}

#' @rdname scale_mnmlab
#' @export
scale_color_mnmlab <- function(palette,...) {
  ggplot2::discrete_scale("color", "mnmlab", mnmlab_pal(palette), ...)
}
#' @rdname scale_mnmlab
#' @export
scale_colour_mnmlab <- scale_color_mnmlab

#' @rdname scale_mnmlab
#' @export
scale_fill_mnmlab <- function(palette,...) {
  ggplot2::discrete_scale("fill", "mnmlab", mnmlab_pal(palette), ...)
}


R/theme_mnmlab.R
#' McCall Lab Theme
#'
#' A theme for [ggplot2] with color blind friendly palettes for use
#' members of the McCall Lab.
#'
#' @seealso [ggplot2::theme]
#' @param text_size Font size for axis text. Legend text will scale by 0.8. Defaults to 17pt.
#' @param title_size Font size for title text. Subtitle will scale by 0.9 and axis title will scale by 0.8. Defaults to 21pt.
#' @param text_color Color for all text in plot. Defaults to black.
#' @param panel_grid_linetype Line type for grid lines in plot. Defaults to 18, a dotted line with 1px on followed by 8px off.
#' @param base_theme Underlying theme to build from. Defaults to linedraw, displaying a margin outline for the plot and gridlines.
#'
#' @examples
#' library(ggplot2)
#' library(palmerpenguins)
#'
#'
#' ggplot(penguins, aes(x = bill_length_mm, y = bill_depth_mm, color = species)) +
#'     scale_color_mnmlab(palette = "martink8") +
#'     theme_mnmlab()
#'
#' @export

theme_mnmlab <- function(
  text_size = 11,
  title_size = 14,
  text_color = "black",
  panel_grid_linetype = "18",
  base_theme = ggplot2::theme_linedraw()
) {
  base_theme + ggplot2::theme(
    text = ggplot2::element_text(
      family = "sans",
      size = text_size,
      color = "black"
    ),
    axis.title = ggplot2::element_text(
      family = "sans",
      size = ggplot2::rel(0.8),
      color = "black"
    ),
    title = ggplot2::element_text(
      family = "serif",
      size = title_size,
      color = "black"
    ),
    plot.subtitle = ggplot2::element_text(
      family = "serif",
      size = ggplot2::rel(0.9),
      color = "grey20",
      #face = "italic"
    ),
    legend.title = ggplot2::element_text(
      family = "sans",
      size = ggplot2::rel(0.8),
      color = "black"
    ),
    legend.text = ggplot2::element_text(
      family = "sans",
      size = ggplot2::rel(0.8),
      color = "black"
    ),
    panel.grid = ggplot2::element_line(
      color = "black",
      linetype = "18",
      size = 4
    ),
    strip.background = element_rect(fill = "#BBBBBB"),
    strip.text = element_text(color = "black")
  )
}


R/theme_mccall_lab.R
#' McCall Lab Theme
#'
#' A theme for [ggplot2] with color blind friendly palettes for use
#' members of the McCall Lab.
#'
#' @seealso [ggplot2::theme]
#' @param text_size Font size for axis text. Legend text will scale by 0.8. Defaults to 17pt.
#' @param title_size Font size for title text. Subtitle will scale by 0.9 and axis title will scale by 0.8. Defaults to 21pt.
#' @param text_color Color for all text in plot. Defaults to black.
#' @param panel_grid_linetype Line type for grid lines in plot. Defaults to 18, a dotted line with 1px on followed by 8px off.
#' @param base_theme Underlying theme to build from. Defaults to linedraw, displaying a margin outline for the plot and gridlines.
#'
#' @examples
#' library(ggplot2)
#' library(palmerpenguins)
#'
#'
#' ggplot(penguins, aes(x = bill_length_mm, y = bill_depth_mm, color = species)) +
#'     scale_color_mccall_lab(palette = "martink8") +
#'     theme_mccall_lab()
#'
#' @export

theme_mccall_lab <- function(
  text_size = 11,
  title_size = 14,
  text_color = "black",
  panel_grid_linetype = "18",
  base_theme = ggplot2::theme_linedraw()
) {
  base_theme + ggplot2::theme(
    text = ggplot2::element_text(
      family = "sans",
      size = text_size,
      color = "black"
    ),
    axis.title = ggplot2::element_text(
      family = "sans",
      size = ggplot2::rel(0.8),
      color = "black"
    ),
    title = ggplot2::element_text(
      family = "serif",
      size = title_size,
      color = "black"
    ),
    plot.subtitle = ggplot2::element_text(
      family = "serif",
      size = ggplot2::rel(0.9),
      color = "grey20",
      #face = "italic"
    ),
    legend.title = ggplot2::element_text(
      family = "sans",
      size = ggplot2::rel(0.8),
      color = "black"
    ),
    legend.text = ggplot2::element_text(
      family = "sans",
      size = ggplot2::rel(0.8),
      color = "black"
    ),
    panel.grid = ggplot2::element_line(
      color = "black",
      linetype = "18",
      size = 4
    )
  )
}


data-raw/paulTol_pals.R
paulTol_pals <- list(highContrast = c("#BB5566",
                                      "#004488",
                                      "#DDAA33"),
                     bright = c("#4477AA",
                                "#66CCEE",
                                "#228833",
                                "#CCBB44",
                                "#EE6677",
                                "#AA3377",
                                "#BBBBBB"),
                     vibrant = c("#0077BB",
                                "#33BBEE",
                                "#009988",
                                "#EE7733",
                                "#CC3311",
                                "#EE3377",
                                "#BBBBBB"),
                     muted = c("#332288",
                               "#88CCEE",
                               "#44AA99",
                               "#117733",
                               "#999933",
                               "#DDCC77",
                               "#CC6677",
                               "#882255",
                               "#AA4499",
                               "#DDDDDD"),
                     pale = c("#BBCCEE",
                              "#CCEEFF",
                              "#CCDDAA",
                              "#EEEEBB",
                              "#FFCCCC",
                              "#DDDDDD"),
                     dark = c("#222255",
                              "#225555",
                              "#225522",
                              "#666633",
                              "#663333",
                              "#555555"),
                     light = c("#77AADD",
                               "#99DDFF",
                               "#44BB99",
                               "#BBCC33",
                               "#AAAA00",
                               "#EEDD88",
                               "#EE8866",
                               "#FFAABB",
                               "#DDDDDD"))


R/scale_mccall_lab.R
#' Color palette for McCall Lab plots

#' 8 color scale created by [Martin Krzywinski's color blind palettes](http://mkweb.bcgsc.ca/colorblind/palettes.mhtml#page-container), adapted from [Wong, B. (2011) Nature Methods 8:441](https://www.nature.com/articles/nmeth.1618).
#' @references
#' Krzywinski, M. Designing for Color Blindess - Martin Krzywinski / Genome Sciences Center. http://mkweb.bcgsc.ca/colorblind/index.mhtml#page-container
#' Wong, B. (2011). Color blindness. In Nature Methods (Vol. 8, Issue 6, p. 441). https://doi.org/10.1038/nmeth.1618

#' @seealso [ggplot2::scale_color_discrete][ggplot2::scale_fill_discrete]
#' @inheritParams ggplot2::discrete_scale
#' @inheritParams scales::manual_pal
#' @param palette= Palette for color scale. Takes on the values martink8,
#'                 martink12, martink15, or martink48. See reference for images.
#' @param ... Further arguments passed to other methods
#' @name scale_mccall_lab
#'

mccall_lab_pal <- function(palette = c("martink8", "martink12", "martink15", "martink48")) {
  if(palette == "martink8"){
    pal <- ggmccalllab::martink_pals[[1]]
  }
  if(palette == "martink12"){
    pal <- ggmccalllab::martink_pals[[2]]
  }
  if(palette == "martink15"){
    pal <- ggmccalllab::martink_pals[[3]]
  }
  if(palette == "martink48"){
    pal <- ggmccalllab::martink_pals[[4]]
  }
  scales::manual_pal(pal)
}

#' @rdname scale_mccall_lab
#' @export
scale_color_mccall_lab <- function(palette = c("martink8", "martink12", "martink15", "martink48"),...) {
  ggplot2::discrete_scale("color", "mccall_lab", mccall_lab_pal(palette), ...)
}
#' @rdname scale_mccall_lab
#' @export
scale_colour_mccall_lab <- scale_color_mccall_lab

#' @rdname scale_mccall_lab
#' @export
scale_fill_mccall_lab <- function(palette = c("martink8", "martink12", "martink15", "martink48"),...) ggplot2::discrete_scale("fill", "mccall_lab", mccall_lab_pal(palette = c("martink8", "martink12", "martink15", "martink48")), ...)
